service-statistics-pxweb/src/main/java/fi/nls/oskari/control/statistics/plugins/pxweb/parser/PrefixedRegionsValueProcessor.java

```diff
@@ -0,0 +1,40 @@
+package fi.nls.oskari.control.statistics.plugins.pxweb.parser;
+
+import fi.nls.oskari.control.statistics.data.RegionValue;
+import fi.nls.oskari.control.statistics.plugins.db.DatasourceLayer;
+import org.json.JSONArray;
+
+import java.util.Optional;
+
+/**
+ * Allows statistical datasource to have a prefix in region ID. If prefix is defined for layer this
+ * processor returns values for ONLY the regions that have the prefix and removes the prefix from the returned region ID.
+ *
+ * Requires datasource layer link to have a config including:
+ *  {
+ *      "valueProcessor": "fi.nls.oskari.control.statistics.plugins.pxweb.parser.PrefixedRegionsValueProcessor",
+ *      "statsRegionPrefix": "[prefix for region ids in statistical data]"
+ *  }
+ */
+public class PrefixedRegionsValueProcessor extends ValueProcessor {
+
+    public Optional<RegionValue> getRegionValue(JSONArray values, String regionId, int valueIndex, DatasourceLayer layer) {
+        Optional<RegionValue> value = super.getRegionValue(values, regionId, valueIndex, layer);
+        if(!value.isPresent()) {
```

Comment:
If you want to have more Optional you could move the `regionIdPrefix == null` check up and then do a 
```
return value.filter(v -> v.getRegion().startsWith(prefix))
    .map(v -> new RegionValue(v -> v.getRegion().substring(prefix.length()), v.getValue()));
```